### PR TITLE
Initial Slider Implementation

### PR DESCRIPTION
**Pull request by bacalj**
This is an initial implementation of the slider.  It includes: 
- There are 48(!) files changed, but many of them are small api changes @kswenson made in the process of refactoring the axis into its own entity. 
- The slider files are under `v3/src/components/slider/`
- Slider uses a V3 Axis with all scale controls
- Thumb can play, and be set manually
- Input is rounded to two decimal places
- There are a number of TODO items in comments - all of which I would like your thoughts on.  I'll flag them on the PR as well. 
- There is also the addition of padding around the slider within the "tile" - left out for now and being saved for "the responsive era" - that said, the axis does stretch /compress as the element does.

### CODE REVIEW

**Review by kswenson — APPROVED** [review]
👍 Looks good -- I started to make a bunch of suggestions but then just decided to add a commit so take a look.

**Comment by kswenson** [comment]
Also, not surprisingly there are a bunch of conflicts, so maybe we tackle those together on Monday.

**Comment by bfinzer** [comment]
I'll leave it to Joe and Kirk to work out the kinks. Let me know if I can help.

**Comment by kswenson** [comment]
I have local branch where I've rebased and addressed all the conflicts. I'll wait until reviews are complete to push the branch.

File: v3/src/components/slider/editable-slider-value.tsx
Comment (bacalj) [inline v3/src/components/slider/editable-slider-value.tsx:15]: Oh, wow. I should have guessed that something like this existed in D3!

File: v3/src/components/slider/editable-slider-value.tsx
Comment (bacalj) [inline v3/src/components/slider/editable-slider-value.tsx:39]: `isFinite`🙏

File: v3/src/components/slider/slider-component.tsx
Comment (bacalj) [inline v3/src/components/slider/slider-component.tsx:70]: @kswenson now that I look at the other changes, should `incrementSliderValue` be a method in the model?

File: v3/src/components/slider/slider-component.tsx
Comment (bacalj) [inline v3/src/components/slider/slider-component.tsx:91]: @kswenson will the inspector always be in a portal, since it will be a single component that will have to appear in disparate places throughout the dom and component tree?

File: v3/src/components/slider/slider-model.ts
Comment (bacalj) [inline v3/src/components/slider/slider-model.ts:37]: this guarantees well set n to a multiple of the `multipleOf`?

File: v3/src/components/slider/slider-component.tsx
Comment (bacalj) [inline v3/src/components/slider/slider-component.tsx:161]: @bacalj this is still reactive, see other side, you need to address property in dependency array, not destructure ahead of time

**Review by bacalj — COMMENTED** [review]
Thanks - I put in some comments that are general questions and things I didn't want to forget.  Works well.

File: v3/src/components/slider/editable-slider-value.tsx
Comment (kswenson) [reply v3/src/components/slider/editable-slider-value.tsx:39]: I generally find `isFinite()` preferable to `isNaN()` both because infinity should usually be handled as invalid along with `NaN` but also because it doesn't lead to constructs like `!isNaN()` which is a double-negative and seems harder to reason about.

**Review by bfinzer — APPROVED** [review]
Looks good. Please be sure a squash on merge.

**Pull request merged**
